lib/commons/dom/is-offscreen.js

```diff
@@ -1,27 +1,26 @@
 /*global dom */
 
-dom.isOffscreen = function (element) {
-	'use strict';
-	var noParentScrolled = function (element, offset) {
-		element = element.parentNode;
-		while (element.nodeName.toLowerCase() !== 'html') {
-			if (element.scrollTop) {
-				offset += element.scrollTop;
-				if (offset >= 0) {
-					return false;
-				}
+function noParentScrolled (element, offset) {
+	element = dom.getComposedParent(element);
+	while (element && element.nodeName.toLowerCase() !== 'html') {
+		if (element.scrollTop) {
+			offset += element.scrollTop;
+			if (offset >= 0) {
+				return false;
 			}
-			element = element.parentNode;
 		}
-		return true;
-	};
+		element = dom.getComposedParent(element);
+	}
+	return true;
+}
 
-	var leftBoundary,
-		docElement = document.documentElement,
-		styl = window.getComputedStyle(element),
-		dir = window.getComputedStyle(document.body || docElement)
-			.getPropertyValue('direction'),
-		coords = dom.getElementCoordinates(element);
+dom.isOffscreen = function (element) {
```

Comment:
The fact that these style changes are mixed up in this pull request makes it much more difficult than it should be to review.